HACKING.rst.txt

```diff
@@ -128,13 +139,50 @@ deprecation message for the changed behavior.  This should be part of the
 review process.
 
 
+Reviewing pull requests
+=======================
+
+Reviewing open pull requests (PRs) is very welcome, and a valuable way to help
+increase the speed at which the project moves forward.  We encourage everyone
+to get involved in the review process; it's also a great way to get familiar
+with the code base.  Reviewers should ask themselves some or all of the
+following questions:
+
+- Was this change adequately discussed (relevant for new features and changes
+  in existing behavior)?
+- Is the feature scientifically sound? Algorithms may be known to work based on
+  literature; otherwise, closer look at correctness is valuable.
+- Is the intended behavior clear under all conditions (e.g. unexpected inputs
+  like empty arrays or nan/inf values)?
+- Does the code meet the quality, test and documentation expectation outline
+  under `Contributing new code`_?
+
+If you have specific knowledge/experience in a particular
+area (say "optimization algorithms" or "special functions") then reviewing PRs
+in that area is especially valuable - sometimes PRs with technical code have to
+wait for a long time to get merged due to a shortage of appropriate reviewers.
```

Comment:
Maybe switch the order: invite domain experts first, and then mention that we welcome everybody to check the style/details.

Optional: maybe add something like "if we do not know you yet, consider introducing yourself".